src/rtos/rtos.c

```diff
@@ -591,7 +591,7 @@ int rtos_get_gdb_reg(struct connection *connection, int reg_num)
 
 		free(reg_list);
 	}
-	return ERROR_FAIL;
+	return ERROR_NOT_IMPLEMENTED;
```

Comment:
Overall, this looks all right to me. I have just two suggestions:

1) Code style: For readability, I'd recommend to use here the early-return code structure, as shown below.

2) What should be returned at the very end of the function, that is if the `reg_num` is not found in `reg_list`? I believe it should be ERROR_FAIL.

```c
int rtos_get_gdb_reg(...)
{
    if ( (!target->rtos) || 
        current_threadid == -1 ||
        current_threadid == 0 ||
        /* ... */ ) {
        /* early return */
        return ERROR_NOT_IMPLEMENTED;
    }

    /* The rest of the algorithm continues here */
    /* ... */
    /* ... */

    /* What should be returned if we reach the end? */
    LOG_ERROR("Did not find register number %d in RTOS register list", reg_num);
    return ERROR_FAIL; 
    
    /* or */

    return ERROR_NOT_IMPLEMENTED;
}
```

Reply:
Done. Thanks.